sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/ReactorNettyHttpResponseBase.java

```diff
@@ -31,8 +42,212 @@ public final String getHeaderValue(String name) {
+            if (name == null) {
+                return this;
+            }
+
+            if (value == null) {
+                remove(name);
+            } else {
+                nettyHeaders.add(name, value);
+            }
+            return this;
+        }
+
+        @Override
+        public HttpHeader get(String name) {
+            if (nettyHeaders.contains(name)) {
+                // Be careful here: Netty's HttpHeaders 'get' method will return only the first value,
+                // which is obviously not what we want to call!
+                // We call 'getAll' instead, but unfortunately there is a representation mismatch:
+                // Netty HttpHeaders uses List<String>, whereas azure-core HttpHeaders joins it all into a
+                // comma-separated String.
+                return new NettyHttpHeader(this, name, nettyHeaders.getAll(name));
+            }
+            return null;
+        }
+
+        @Override
+        public HttpHeader remove(String name) {
+            HttpHeader header = get(name);
+            nettyHeaders.remove(name);
+            return header;
+        }
+
+        @Override
+        public String getValue(String name) {
+            return nettyHeaders.get(name);
+        }
+
+        @Override
+        public String[] getValues(String name) {
+            return nettyHeaders.getAll(name).toArray(new String[] { });
+        }
+
+        @Override
+        public Map<String, String> toMap() {
+            if (abstractMap == null) {
+                abstractMap = new AbstractMap<String, String>() {
+                    @Override
+                    public int size() {
+                        return nettyHeaders.size();
+                    }
+
+                    @Override
+                    public boolean isEmpty() {
+                        return nettyHeaders.isEmpty();
+                    }
+
+                    @Override
+                    public boolean containsKey(Object key) {
+                        return nettyHeaders.contains((String) key);
+                    }
+
+                    @Override
+                    public boolean containsValue(Object value) {
+                        throw new UnsupportedOperationException();
+                    }
+
+                    @Override
+                    public String get(final Object key) {
+                        // Calling nettyHeaders.get(key) returns only the first value in the headers for the given key.
+                        // If there are multiple values, the user not get the result they expect.
+                        // For now, this is resolved by joining the headers back into a String here, with the obvious
+                        // performance implication, and therefore it is recommended that users steer away from calling
+                        // httpHeaders.toMap().get(key), and instead be directed towards httpHeaders.get(key), where
+                        // caching is available.
```

Comment:
Is this accurate? I don't think `httpHeaders.get(key)` is doing any caching either as it creates a new NettyHttpHeader instance each time `get()` is called.

Reply:
I've clarified the text.